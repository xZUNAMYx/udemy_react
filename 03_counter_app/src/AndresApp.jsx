import PropTypes from "prop-types";

const getName = (name)=>{
    return name;
}

export const AndresApp = ({title, subTitle, name}) =>{
    return (
    //   Sinonimo del fragmento
        <>  
            <h1 data-testid='test-title'> {title} </h1>
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    )
}

AndresApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
}

AndresApp.defaultProps = {
    // title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
    name: 'Andres Alvarez',
}