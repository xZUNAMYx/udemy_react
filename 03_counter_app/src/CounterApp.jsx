import { useState } from "react";

import PropTypes from "prop-types";

export const CounterApp = ({value})=>{
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () =>{
        setCounter( counter +1 );
        // setCounter( (c) => c +1 );
    }

    const handleSubtract = () =>{
        setCounter( counter - 1 );
    }

    const handleReset = () =>{
        setCounter( value );
        // setCounter( (c) => c +1 );
    }
    
    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={(event) => {handleAdd(event)}}>
                +1
            </button>
            <button onClick={(event) => {handleSubtract(event)}}>
                -1
            </button>
            <button onClick={(event) => {handleReset(event)}}>
                Reset
            </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}