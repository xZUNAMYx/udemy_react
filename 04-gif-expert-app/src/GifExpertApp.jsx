import { useState } from "react";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One punch', 'Luffy']);
    console.log(categories);

    const onAddCategory = ()=>{
        setCategories([...categories, 'One Piece']); // una forma de hacerlo
        // setCategories( (category)=>[...category, 'One Piece']); // Otra forma de hacerlo
    }

    return (
        <>
            {/* // TÃ­tulo */}
            <h1>GifExpertApp</h1>

            {/* // Input */}

            {/* // Listado de Gif */}
            <button onClick={(event)=>onAddCategory(event)}>Agregar</button>
            <ol>
                {categories.map( (category)=>{
                    return <li key={category}>{category}</li>
                })}
            </ol>
                {/* // Git Item */}
        </>
        
           
    );
}
