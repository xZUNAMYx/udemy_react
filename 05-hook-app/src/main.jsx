import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
// import { HooksApp } from './HooksApp.jsx'
// import { CounterApp } from './01-useState/CounterApp.jsx'
// import { CounterWithCustomHook } from './01-useState/CounterWithCustomHook.jsx'
// import { SimpleForm } from './02-useEffect/SimpleForm.jsx'
// import { FormWithCustomHook } from './02-useEffect/FormWithCustomHook.jsx'
// import { FocusScreen } from './04-useRef/FocusScreen.jsx'
// import { Memorize } from './06-memos/Memorize.jsx'
// import { MemoHook } from './06-memos/MemorizeHooks.jsx'
// import { CallbackHook } from './06-memos/CallbackHook.jsx'
// import { Padre } from './07-tarea-memo/Padre.jsx';

// import "./08-useReducer/intro-reducer.js";

// import { TodoApp } from './08-useReducer/TodoApp';
import {BrowserRouter} from "react-router-dom";
import './index.css';
import { MainApp } from './09-useContext/MainApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <BrowserRouter>
        {/* <React.StrictMode> */}
        <MainApp />
        {/* </React.StrictMode>, */}
    </BrowserRouter>
)
