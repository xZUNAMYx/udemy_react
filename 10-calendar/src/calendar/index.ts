export * from './components/CalendarEvent';
export * from './components/CalendarModal';
export * from './components/Navbar';
export * from './components/ButtonAddNew';
export * from './components/ButtonDelete';
    
export * from './pages/CalendarPage';