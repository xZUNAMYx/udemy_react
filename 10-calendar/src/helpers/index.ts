export * from '../helpers/CalendarLocalizer';
export * from '../helpers/GetMessages';