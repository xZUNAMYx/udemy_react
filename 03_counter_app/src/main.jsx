import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from "./CounterApp";
import { AndresApp } from "./AndresApp";

import './styles.css'
// import { CounterApp } from './CounterApp';

ReactDOM.createRoot( document.getElementById('root') ).render(

    <React.StrictMode>
        {/* <AndresApp title='Hola, soy Andres' subTitle='Mis primeros PropTypes'/> */}
        <CounterApp value={0}/>
        {/* <AndresApp /> */}
    </React.StrictMode>

);