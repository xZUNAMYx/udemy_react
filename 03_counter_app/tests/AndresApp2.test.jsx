import { render, screen } from "@testing-library/react";
import { AndresApp } from "../src/AndresApp";

describe('Pruebas en <AndresApp2 />', () => { 
    const title = 'Hola soy Andres Alvarez';
    const subTitle = 'Soy un subtitulo';

    test('Debe hacer match con el snapShot', () => { 
        const {container} = render( <AndresApp title={title}/>);

        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el mensaje "Hola soy Andres ALvarez"', () => { 
        render( <AndresApp title={title}/>);
        expect(screen.getByText(title)).toBeTruthy();
        // screen.debug();
    });

    test('Debe mostrar el titulo en un h1', () => { 
        render( <AndresApp title={title}/>);
        expect( screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
    });

    test('Debe mostrar el subtitulo enviado por props', () => { 
        render( 
            <AndresApp 
                title={ title } 
                subTitle={ subTitle }
            /> 
        );

        expect(screen.getAllByText(subtitle).length).toBe(2);
    });
});