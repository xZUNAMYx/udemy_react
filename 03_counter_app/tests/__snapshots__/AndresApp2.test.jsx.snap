// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AndresApp2 /> Debe hacer match con el snapShot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy Andres Alvarez
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Andres Alvarez
  </p>
</div>
`;
